text/dependencies/rfcs/0000-dependency-management-phase-two.md

```diff
@@ -0,0 +1,195 @@
+# Dependency Management Phase 2: Workflow and Github Action Generalization RFC
+
+## Proposal
+
+The process around dependency version discovery, retrieval, possible
+compilation, and metadata storage should be carried out through a set of
+Dependencies-team maintained Github actions and workflows. The workflows will
+leverage dependency-specific code outlined in [RFC 0000: Phase
+1](https://github.com/paketo-buildpacks/rfcs/blob/dependency-management-step-one/text/dependencies/rfcs/0000-dependency-management-phase-one.md)
+and will live in a common location to be reused across buildacks.
+
+## Motivation
+
+Per top-level [RFC 0000:
+Overview](https://github.com/paketo-buildpacks/rfcs/blob/dependency-management-top-level/text/dependencies/rfcs/0000-dependency-management-overview.md),
+the second phase of the dependency management process update will be to
+completely rework the existing set of actions and workflows from the
+[dep-server](https://github.com/paketo-buildpacks/dep-server) such that the
+dep-server is no longer need because the entire process runs alongside the
+buildpacks.
+
+This phase will address concerns around the opaque complexity of the dep-server
+process. Additionally, it addresses the painpoint of maintainability.
+Dependencies maintainers will own the new workflows and actions, ensuring every
+relevant buildpack follows the same overall system architecture, while giving
+buildpacks maintainers freedom to make dependency-specific choices separately
+and debug issues with full authority.
+
+The new actions will also be simple to run locally in the event of
+Github Actions degradations, making the dependency-management system more
+flexible in use and transparent in process.
+
+## Detailed Explanation
```

Comment:
Buildpack authors will need to specify the desired operating systems somehow. This could be specified in a file such as `<buildpack>/dependency/operating_systems.json` that looks like this:

```json
["ubuntu-18.04", "ubuntu-22.04", "macos-latest"]
```

Note that it's currently written on a single line (see https://github.com/joshuatcasey/bundler/runs/6802820650?check_suite_focus=true) but that should be fixable.

Reply:
Do we need to give some thought to those OS/arch combinations that are not available natively via github actions virtual environments (https://github.com/actions/virtual-environments/)?

UBI is built on RHEL, are we sure that binaries built on Ubuntu will work for them? https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image

Reply:
Yea, this is something I had looked into, I think we might be able to do something with the base image used in the Github Action Dockerfile

Reply:
addressed in the latest version of the rfc